### PR TITLE
🛠️ `Neighborhood`: Fail CI if there are lint violations

### PR DESCRIPTION
**Pull request by anaulin**
https://github.com/zinc-collective/convene/issues/1156

Adds a lint-check step to CI, in parallel with the test runs, that runs:
* Ruby style check
* JS and CSS style check

I tried adding a check that the ERD is up-to-date, but that requires setting up the database in the lint-check step and also installing graphviz in CI, which seems a little heavyweight. A pre-commit check might be better for the ERD check.

image

### CODE REVIEW

**Comment by zspencer** [comment]
How would you feel about a `rubocop -a` as part of this to cut down on the `todo`?

File: .github/workflows/test-convene-web.yml [hunk 1]
```diff
@@ -172,3 +172,19 @@ jobs:
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Ruby and install gems
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+      - run: bundle exec rubocop --parallel --config .rubocop.yml
```
Comment (zspencer) [inline .github/workflows/test-convene-web.yml:190]: Do we want one for `prettier` as well? That's what is (ostensibly) formatting our `css` and `js`?

**Review by zspencer — APPROVED** [review]
Love it! Thank you!

File: .github/workflows/test-convene-web.yml (hunk 1 above)
Comment (anaulin) [reply .github/workflows/test-convene-web.yml:199]: Added.

**Comment by anaulin** [comment]
@zspencer this is ready for review now

**Review by zspencer — APPROVED** [review]
I love it!!!

**Pull request merged**